src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -312,18 +313,21 @@ internal bool CommitUnsynchronized()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal void Advance(int bytes)
         {
-            if ((uint)bytes > (uint)_writingHeadMemory.Length)
+            lock (_sync)
```

Comment:
You can remove the AggressiveInlining.

Reply:
cc @benaadams sorry dude, locks mean no more inlining 😆

Reply:
So much lock for writer that doesn't support concurrency 😢